src/listensocket.cpp

```diff
@@ -29,6 +29,18 @@ ListenSocket::ListenSocket(ConfigTag* tag, const irc::sockets::sockaddrs& bind_t
 	: bind_tag(tag)
 	, bind_sa(bind_to)
 {
+	// Are we creating a UNIX socket?
+	const std::string path = tag->getString("path");
+	if (!path.empty())
+	{
+		// Expand the path relative to the config directory.
+		const std::string fullpath = ServerInstance->Config->Paths.PrependData(path);
+		// Is 'replace' enabled?
+		const bool replace = tag->getBool("replace");
+		if (replace && irc::sockets::isunix(fullpath))
+			remove(fullpath.c_str());
```

Comment:
```suggestion
			unlink(fullpath.c_str());
```

(This was a bug in the original code).